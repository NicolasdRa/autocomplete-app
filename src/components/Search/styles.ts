import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
	createStyles({
		textField: {
			width: '16rem',
			marginRight: '.5rem',

			[theme.breakpoints.down('sm')]: {
				width: '100%',
				marginRight: 0,
				marginBottom: '1rem',
			},
		},
	}),
);
