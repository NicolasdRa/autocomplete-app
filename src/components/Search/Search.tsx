import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import IconButton from '@material-ui/core/IconButton';
import SearchIcon from '@material-ui/icons/Search';

import { useStyles } from './styles';

interface Props {
	handleChangeQuery: (e: unknown) => void;
}

export const Search: React.FC<Props> = ({ handleChangeQuery }) => {
	const classes = useStyles();

	return (
		<Paper component='form' className={classes.root}>
			<InputBase
				className={classes.input}
				placeholder='Search product'
				inputProps={{ 'aria-label': 'search google maps' }}
				onChange={handleChangeQuery}
				defaultValue=''
			/>
			<IconButton
				type='submit'
				className={classes.iconButton}
				aria-label='search'>
				<SearchIcon />
			</IconButton>
		</Paper>
	);
};
