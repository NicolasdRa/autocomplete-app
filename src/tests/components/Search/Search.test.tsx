import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { Search } from '../../../components/Search/Search';

const handleChangeQuery = jest.fn();

describe('tests on <Search/>', () => {
	const wrapper = mount(<Search handleChangeQuery={handleChangeQuery} />);

	test('should render <Search/> correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should call handleChangeQuery', () => {
		const input = wrapper.find('input');

		input.simulate('change');
		expect(handleChangeQuery).toHaveBeenCalled();
	});
});
