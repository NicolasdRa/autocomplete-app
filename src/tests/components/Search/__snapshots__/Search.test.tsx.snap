// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Search/> should render <Search/> correctly 1`] = `
<div
  className="MuiFormControl-root makeStyles-textField-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
    data-shrink={false}
    htmlFor="outlined-adornment-password"
  >
    Search product
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
    onClick={[Function]}
  >
    <input
      aria-describedby="Search product"
      aria-invalid={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-3"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;
